### ISSUE TITLE
bs-select error using track by 

### ISSUE THREAD

**Issue by lars159** [mgcrea/angular-strap#1667]
I having a hard time getting the bs-select working with track by or 'as'
heres a plunker showing some error some of my errors.

http://plnkr.co/edit/tWVK3KGSzaY0taV0wTVi?p=preview

**Comment by vmlf01** [comment]
If you use the multiple select, the ng-model needs to be an array. But even after making that change, you get some errors when specifying an object as the initial value in the array, which you shouldn't get.

I think this is a more general problem of initializing the bs-select with an object reference that is not the same as the options, even though "we know" that they refer to the same thing. I think this as come up on the issues list before.

Here is an updated plunker that works, maybe it will help you:
http://plnkr.co/edit/weK2GOILFnJRJCPFYqjD?p=preview

**Comment by lars159** [comment]
Yes it's more of a general problem of initializing. Your answer was very helpful thanks

**Comment by polo2ro** [comment]
I have added a pull request and unit test for my own bug, more or less similar to this one : 
https://github.com/mgcrea/angular-strap/pull/1818

it is about initializing a ng-model with a value not present in options

**Comment by Routhinator** [comment]
It's not just multiple selects. Regular selects ignore the track by option and use the label instead.
